work/sixth/index.html

```diff
@@ -16,6 +16,72 @@
   <script src="../../common/js/jquery.js"></script>
   <script>
     // 回答はここに書いてね
+    let encountHtml =
+      `
+        <p class="name"></p>
+        <div>
+          <img class="img" src="" alt="">
+        </div>
+        <div>
+          <button class="btn" id="catch">捕まえる</button>
+          <button class="btn" data-js="reset">逃げる</button>
+        </div>
+      `
+      
+    let failHtml =
+      `
+        <p class="text">逃げられてしまった...</p>
+        <button class="btn mt" data-js="reset">戻る</button>
+      `
+
+    let doneHtml =
+      `
+        <p class="name">ゲットしました</p>
+        <div>
+          <img class="img" src="" alt="">
+        </div>
+        <button class="btn" data-js="reset">戻る</button>
+      `
+
+    $(function() {
+      let defaultHtml = $('.wrapper').clone(true);
+      function resetHtml() {
+        $(document).on('click', '.btn[data-js]', function() {
+          $('.wrapper').replaceWith(defaultHtml.clone());
+        })
+      };
```

Comment:
[FYI]
こちら、戻るボタンのクリックイベント登録を関数化し複数の箇所で呼び出していますが、イベント登録はonメソッドを使っていればページ読み込み時の一回でイベント付与できますので、何度も呼び出す必要はありません！

そのため、`on('click', クリック対象, コールバック関数)メソッド`で「クリックイベントを付与する」という処理は、何度も呼び出す必要がないので、関数化してもあまり意味がないです！🔍

今回は、クリックイベント自体を関数化するよりは、クリックイベントの時に実行するコールバック関数の方を関数化してあげるのがおすすめです。

```suggestion
      // イベント登録は関数化しない
      $(document).on('click', '.btn[data-js]', resetHtml)
      // イベント時の処理の方は、関数化する
      function resetHtml() {
          $('.wrapper').replaceWith(defaultHtml.clone());
      };
```

Reply:
こちらのコミットにてイベントの処理内容を関数化してクリックイベント時に関数resetHtmlを実行するような記述に変更いたしました。
7678d68686b9abdcdf166891420bd5fe70975a09